### PR TITLE
Fix: return error when credential contains invalid revocation settings

### PR DESCRIPTION
**Pull request by bobhageman**

### CODE REVIEW

File: schemes.go [hunk 1]
```diff
@@ -1252,13 +1252,30 @@ func (scheme *SchemeManager) parseCredentialsFolder(conf *Configuration, issuer
...
+			if attr.RevocationAttribute {
+				revAttrs = true
+			}
 		}
+
+		if len(cred.RevocationServers) == 0 && revAttrs {
+			return errors.New("Revocation attribute specified, but RevocationServer(s) is missing")
+		}
+		if len(cred.RevocationServers) > 0 && !revAttrs {
+			return errors.New("RevocationServer(s) specified, but revocation attribute: '<Attribute revocation=\"true\" />' is missing")
```
Comment (sanderhollaar) [inline schemes.go:1276]: remove parenthesis in `RevocationServer(s)`

File: schemes.go [hunk 2]
```diff
@@ -1252,13 +1252,30 @@ func (scheme *SchemeManager) parseCredentialsFolder(conf *Configuration, issuer
...
+				return errors.New(fmt.Sprintf("Attribute %s.%s cannot contain revocation=\"true\". This needs to be a separate attribute: <Attribute revocation=\"true\" />", attr.IssuerID, attr.ID))
+			}
+
+			if attr.RevocationAttribute {
+				revAttrs = true
+			}
 		}
+
+		if len(cred.RevocationServers) == 0 && revAttrs {
+			return errors.New("Revocation attribute specified, but RevocationServer(s) is missing")
```
Comment (sanderhollaar) [inline schemes.go:1273]: remove parenthesis in `RevocationServer(s)`

File: schemes.go (hunk 1 above)
Comment (ivard) [reply schemes.go:1276]: Is the issue the other way around also handled somewhere? So that would be `len(cred.RevocationServers) == 0 && revAttrs`.

Comment (bobhageman) [reply schemes.go:1276]: @ivard: yes it is. Right above it actually, at line [1272](https://github.com/privacybydesign/irmago/pull/316/files#diff-d9a0f717b4668b699fae9ed83cd3c40b204b233359281a673934659068d53fa8R1272)

**Review by sanderhollaar — APPROVED** [review]

File: schemes.go (hunk 1 above)
Comment (ivard) [reply schemes.go]: Ah of course

**Review by ivard — APPROVED** [review]

**Pull request merged**